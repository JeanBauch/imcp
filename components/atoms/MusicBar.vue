<template>
	<div>
		<div class="flex justify-between">
			<p class="text-sm text-primary-beige-700 font-semibold">
				01:28
			</p>
			<p class="text-sm text-primary-beige-700 font-semibold">
				05:00
			</p>
		</div>
		<div class="flex-1 bg-primary-beige-300 h-2 rounded-3xl">
			<div class="relative bg-primary-beige-700 h-2 rounded-3xl" :style="{ width: `${progressMusic}%` }" />
		</div>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	props: {
		progressMusicPlayed: { type: Number, required: true },
	},
	computed: {
		progressMusic () {
			return this.progressMusicPlayed;
		},
	},
});
</script>
