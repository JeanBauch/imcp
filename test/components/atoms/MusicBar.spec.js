import { mount } from '@vue/test-utils';

import MusicBar from '~/components/atoms/MusicBar.vue';

describe('Components:atoms:MusicBar', () => {
	describe('Snapshots', () => {
		it('should render component with correctly props', () => {
			const wrapper = mount(MusicBar, {
				propsData: {
					progressMusicPlayed: 18,
				},
			});

			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
