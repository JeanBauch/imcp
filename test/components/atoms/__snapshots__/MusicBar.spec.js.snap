// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:MusicBar Snapshots should render component with correctly props 1`] = `
"<div>
  <div class=\\"flex justify-between\\">
    <p class=\\"text-sm text-primary-beige-700 font-semibold\\">
      01:28
    </p>
    <p class=\\"text-sm text-primary-beige-700 font-semibold\\">
      05:00
    </p>
  </div>
  <div class=\\"flex-1 bg-primary-beige-300 h-2 rounded-3xl\\">
    <div class=\\"relative bg-primary-beige-700 h-2 rounded-3xl\\" style=\\"width: 18%;\\"></div>
  </div>
</div>"
`;
